drivers/SmartThings/matter-appliance/src/matter-dishwasher/init.lua

```diff
@@ -0,0 +1,316 @@
+-- Copyright 2023 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+local clusters = require "st.matter.clusters"
+local embedded_cluster_utils = require "embedded-cluster-utils"
+
+local log = require "log"
+
+local DISHWASHER_DEVICE_TYPE_ID = 0x0075
+local version = require "version"
+if version.api < 10 then
+  clusters.DishwasherAlarm = require "DishwasherAlarm"
+  clusters.DishwasherMode = require "DishwasherMode"
+  clusters.OperationalState = require "OperationalState"
+  clusters.TemperatureControl = require "TemperatureControl"
+end
+
+local OPERATIONAL_STATE_COMMAND_MAP = {
+  [clusters.OperationalState.commands.Pause.ID] = "pause",
+  [clusters.OperationalState.commands.Stop.ID] = "stop",
+  [clusters.OperationalState.commands.Start.ID] = "start",
+  [clusters.OperationalState.commands.Resume.ID] = "resume",
+}
+
+local supportedTemperatureLevels = {}
+local dishwasherModeSupportedModes = {}
+
+-- helper functions
+local function key_exists(array, key)
+  for k, _ in pairs(array) do
+    if k == key then
+      return true
+    end
+  end
+  return false
+end
```

Comment:
couldn't you just do

```
if array[key] ~= nil
```

? Also I don't think `array` is the right name here, since that (in lua) implies a map with integer keys starting from 1.

Reply:
Done in [39cdb6f](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1422/commits/39cdb6f5aa61b8bcd2614de83c774e438afa8a3d)